### PR TITLE
Remaining enhancements to complete beta (#712)

### LINKED ISSUE #712: [Pledge] Remaining enhancements to complete beta
### What needs to be completed to remove the beta label from the pledge dashboard:

- [ ] Move the carbon assets card to the bottom
- [ ] Only show the assets with a history to graph (assets that have never been owned will be hidden)
- [x] De-capitalize the card titles, that's too much weight for that specific title
- [ ] Hide image profile placeholder is the user does not have one
- [ ] Change Footprint graph colour scheme to support accessibility for colour blind users. HEX values available in the [figma design](https://www.figma.com/file/moDUIfGjQaISUGDjLxyko4/Multi-Wallet-Pledge?node-id=9%3A621)
Screen Shot 2022-10-01 at 9 55 53 AM

### PR DESCRIPTION
**Pull request by 0xemc**
## Description
Complete the following tasks on the PledgeDashboard

- [x] Move the carbon assets card to the bottom
- [ ] Only show the assets with a history to graph (assets that have never been owned will be hidden) 
^ **Since fetching transaction history per contract was not a simple task a compromise here was to only show assets with balance > 0**
- [x] Hide image profile placeholder is the user does not have one
- [x] Change Footprint graph colour scheme to support accessibility for colour blind users. HEX values available in the [figma design](https://www.figma.com/file/moDUIfGjQaISUGDjLxyko4/Multi-Wallet-Pledge?node-id=9%3A621)

**Note:**
* Please test with number of categories greater than 4 (for the FootprintCard and it's graph)
* @jabby09 font sizes seem larger in the app than in designs

## Related Ticket

Resolves #712 

## Changes

| Before  | After  |
|---------|--------|
|image|image|

## Checklist

- [x] Building the site with `npm run build-site` works without errors
- [x] Building the app with `npm run build-app` works without errors
- [x] I formatted JS and TS files with running `npm run format-all`

### CODE REVIEW

File: site/components/pages/Pledge/PledgeDashboard/Cards/AssetBalanceCard/index.tsx
Comment (0xAeterno) [inline site/components/pages/Pledge/PledgeDashboard/Cards/AssetBalanceCard/index.tsx:19]:
```suggestion
import toNumber from "lodash/toNumber";
```
We can treeshake this function import to prevent the bundle from including the entire lodash module.
Also would you be able to group this import with the other lodash imports above? :)

File: site/components/pages/Pledge/PledgeDashboard/Cards/FootprintCard/index.tsx
Comment (0xAeterno) [inline site/components/pages/Pledge/PledgeDashboard/Cards/FootprintCard/index.tsx:26]: @theprofessora - we currently support a maximum of 10 different footprint categories on pledges.
Are you happy to reuse the colours here or would you extend the palette to include 10 unique colours?

**Review by 0xAeterno — CHANGES_REQUESTED** [review]

File: site/components/pages/Pledge/PledgeDashboard/Cards/FootprintCard/index.tsx
Comment (theprofessora) [reply site/components/pages/Pledge/PledgeDashboard/Cards/FootprintCard/index.tsx:26]: @0xAeterno @Atmosfearful 

I created a 10-color palette for all the categories here: https://www.figma.com/file/moDUIfGjQaISUGDjLxyko4?node-id=1:3#286886177

Our previous color palette of varying shades of green was not WCAG accessible, and part of our goal going forward is to maximize accessibility across the product

Comment (MColl92) [reply site/components/pages/Pledge/PledgeDashboard/Cards/FootprintCard/index.tsx]: Have updated the palette with the new colors. It is also possible to generate a multi-color gradient and then select points within that if we don't want to be limited to 10 colors (though there could be contrast issues once we get to a large number of categories)

File: site/components/pages/Pledge/PledgeDashboard/Cards/AssetBalanceCard/index.tsx
Comment (0xAeterno) [reply site/components/pages/Pledge/PledgeDashboard/Cards/AssetBalanceCard/index.tsx]: Thanks!
Great suggestion - I haven't come across that plugin before but can see the benefits in standardising imports across the repo. I'm not opposed to it.

@Atmosfearful what are your thoughts?

File: site/components/pages/Pledge/PledgeDashboard/Cards/FootprintCard/index.tsx
Comment (0xAeterno) [reply site/components/pages/Pledge/PledgeDashboard/Cards/FootprintCard/index.tsx]: Definitely a good idea to explore if we plan to extend pledge footprint categories past 10.
Might not be the right time with defined limits and accessibility requirements.

**Comment by 0xAeterno** [comment]
image
@theprofessora do you have any ideas on how we might be able to utilise this card if a wallet has no holdings or transaction history? Is there any value in showing the card at all?

**Comment by theprofessora** [comment]
for now we can just leave it empty with a textbox that says "_No carbon assets to show._" (same font size as the text on the pledge card)

However, we're working on a secret project that will make sense to link here once it's live. So I'd say leave the possibility of adding in a CTA on empty carbon asset cards in the near future. But for now a simple empty state can be placeholder

File: site/components/pages/Pledge/PledgeDashboard/Cards/FootprintCard/index.tsx
Comment (MColl92) [reply site/components/pages/Pledge/PledgeDashboard/Cards/FootprintCard/index.tsx:54]: Ah I see, @theprofessora does the order of colour in the graph matter? From the designs it looks like it should progress backwards with the largest asset being blue? (otherwise pink wins out)

image

File: site/components/pages/Pledge/PledgeDashboard/Cards/FootprintCard/index.tsx
Comment (MColl92) [reply site/components/pages/Pledge/PledgeDashboard/Cards/FootprintCard/index.tsx:57]: Fair enough 👍

**Comment by MColl92** [comment]
> for now we can just leave it empty with a textbox that says "_No carbon assets to show._" (same font size as the text on the pledge card)
> 
> However, we're working on a secret project that will make sense to link here once it's live. So I'd say leave the possibility of adding in a CTA on empty carbon asset cards in the near future. But for now a simple empty state can be placeholder

Placeholder added 👍

**Comment by 0xAeterno** [comment]
@Atmosfearful can we authorise this pr to be deployed onto preview build?

File: site/components/pages/Pledge/PledgeDashboard/Cards/FootprintCard/index.tsx
Comment (0xAeterno) [reply site/components/pages/Pledge/PledgeDashboard/Cards/FootprintCard/index.tsx:54]: @MColl92 
Hmm I find the example confusing to reason about. In the original implementation, the palette was defined with colours mapping from largest to smallest footprint.
So if we call `#calculateFootprintPercent` on line 73, the colours will be mapped according to the index of already sorted categories.
https://github.com/KlimaDAO/klimadao/blob/0d2139db08c4563536332ba4f78a5e102baa908e/site/components/pages/Pledge/PledgeDashboard/Cards/FootprintCard/index.tsx#L70-L76

So it's redundant to sort again and calculate the palette index. We just need to ensure the colour palette defined is in the order we expect. I think it is worth revising the order of the colours, pink does look rather funny. It'd be much nicer to be on theme with green as the largest category.

Hope that makes sense!

File: site/components/pages/Pledge/PledgeDashboard/Cards/AssetBalanceCard/index.tsx
Comment (0xAeterno) [inline site/components/pages/Pledge/PledgeDashboard/Cards/AssetBalanceCard/index.tsx:73]: This can be written below as:
```ts
import { Trans } from `lingui/macro`;

<Text t="body2">
  <Trans id="pledges.dashboard.assets.emptyBalance">
    No carbon assets to show.
  </Trans>
</Text>
```

File: site/components/pages/Pledge/PledgeDashboard/Cards/AssetBalanceCard/index.tsx
Comment (0xAeterno) [inline site/components/pages/Pledge/PledgeDashboard/Cards/AssetBalanceCard/index.tsx:116]:
```suggestion
        {tokenHoldingAndBalances.length === 0 && (
          <Text t="body2">{emptyBalanceText}</Text>
        )}
```

**Comment by jabby09** [comment]
@0xAeterno @MColl92 looks like you guys are making good progress. Is there an ETA on when this might be ready for QA/testing? thanks

File: site/components/pages/Pledge/PledgeDashboard/Cards/FootprintCard/index.tsx
Comment (MColl92) [reply site/components/pages/Pledge/PledgeDashboard/Cards/FootprintCard/index.tsx:54]: Yep I agree the logic is confusing. 

Since the sort on line 71 is from smallest to largest then no matter what order we define the `COLOR_PALLETTE` the "largest" (or `categories.length`) index value will always be variable (so no guarantee of the largest color).

For now I will reverse the array to ensure that smallest value is always green and we can revisit later?

File: site/components/pages/Pledge/PledgeDashboard/Cards/AssetBalanceCard/index.tsx
Comment (MColl92) [reply site/components/pages/Pledge/PledgeDashboard/Cards/AssetBalanceCard/index.tsx:73]: Much cleaner👍

File: site/components/pages/Pledge/PledgeDashboard/Cards/AssetBalanceCard/index.tsx
Comment (Atmosfearful) [reply site/components/pages/Pledge/PledgeDashboard/Cards/AssetBalanceCard/index.tsx]: #757

File: site/components/pages/Pledge/PledgeDashboard/Cards/AssetBalanceCard/index.tsx
Comment (Atmosfearful) [inline site/components/pages/Pledge/PledgeDashboard/Cards/AssetBalanceCard/index.tsx:76]: built in js `Number` should work fine `!!Number(balance)`

**Review by Atmosfearful — APPROVED** [review]
Looks great to me, I will wait for the others to approve before I merge @theprofessora  @0xAeterno

File: site/components/pages/Pledge/PledgeDashboard/Cards/AssetBalanceCard/index.tsx
Comment (MColl92) [reply site/components/pages/Pledge/PledgeDashboard/Cards/AssetBalanceCard/index.tsx]: Updated

File: site/components/pages/Pledge/PledgeDashboard/Cards/FootprintCard/index.tsx
Comment (0xAeterno) [reply site/components/pages/Pledge/PledgeDashboard/Cards/FootprintCard/index.tsx]: Let's wait for @theprofessora to make the call on how he'd like to apply the colour palette.
We've defined a few potential ways to approach it based on his call tho :D
IMO I don't think this is a showstopper though, we can always make a ticket and revisit this.

**Review by theprofessora — APPROVED** [review]
looks good, i like how we solved the category color issue. let's ship it!

File: site/components/pages/Pledge/PledgeDashboard/Cards/FootprintCard/index.tsx
Comment (0xemc) [reply site/components/pages/Pledge/PledgeDashboard/Cards/FootprintCard/index.tsx]: Sure, happy to close and come back to this. Up to you guys 👍

**Review by 0xAeterno — APPROVED** [review]

**Comment by 0xemc** [comment]
@Atmosfearful Seems I might need you to authorise the Vercel deploys

**Pull request merged**
